n3fit/src/n3fit/model_trainer.py

```diff
@@ -854,6 +863,11 @@ def hyperparametrizable(self, params):
 
         # Generate the grid in x, note this is the same for all partitions
         xinput = self._xgrid_generation()
+        
+        # compute photon:
+        photon=Photon(theoryid=self.theoryid, fiatlux_runcard=self.fiatlux_runcard)
+        photon_array = photon.photon_fitting_scale(xinput.input_l.tensor_content[0,:,0])
```

Comment:
The `Photon` instance survives only for a single line.
Why not to replace the class with a single function with three parameters? (instead of two functions with respectively two and one?)

Of course, if the goal is modularization, the function can call the other two functions internally.
But three parameters don't seem too many to me...

Reply:
The photon instance will have to be used a few more times later on (potentially infinity, but in practice for the sum rules and the output)

Reply:
Ok, so I'll take my considerations for the final product.